Week-7 Wechaty Polyglot (#1013)
wechaty/wechaty.js.org: docusaurus/docs/polyglot/overview.mdx

<PolyglotTable />

## Wechaty Getting Started Project

> If you are a total beginner to Wechaty, this project is the best starting point for you. You can run it on a Cloud IDE in a couple of steps or on a locally on your machine.

Reviewer: ```suggestion
If you are a total beginner to Wechaty, this project is the best starting point for you. You can run it on a Cloud IDE in a couple of steps or locally on your machine.
```

**Features of Wechaty Getting Started project**

> * It works in all operating systems like on Linux, Mac or Windows.

Reviewer: ```suggestion
* It works in all operating systems like Linux, Mac, or Windows.
```

* It Supports all puppets like Web, Pad, Windows, and Mac.
* It replies with a `dong` message whenever it receives a `ding` message.

> ### Running Wechaty Getting Started Project on a cloud based IDE

Reviewer: ```suggestion
### Running Wechaty Getting Started Project on a cloud-based IDE
```

> The fastest way to get started with Wechaty is to use a Cloud based IDE for running the Wechaty Getting Started Project. You can either use Gitpod or Google Cloud Shell.
> If you are a total beginner, then we recommed Gitpod.

Reviewer: ```suggestion
The fastest way to get started with Wechaty is to use a Cloud-based IDE for running the Wechaty Getting Started Project. You can either use Gitpod or Google Cloud Shell.
If you are a total beginner, then we recommend Gitpod.
```

#### Using Gitpod Wechaty

[![GitPod Ready-to-Code][gitpod_img]][gitpod_link]

[gitpod_img]: https://img.shields.io/badge/Gitpod-Ready--to--Code-blue?logo=gitpod
[gitpod_link]: https://gitpod.io/#https://github.com/wechaty/wechaty-getting-started

> Gitpod  is an online and open source platform for automated and ready-to-code development environments.If you have never used gitpod before, you will be required to login using your gitHub account.
> You can learn more about Gitpod  Wechaty from our blog: [Getting Started Without Leaving Your Browser: Wechaty Gitpod, @huan, Feb 06, 2021](https://wechaty.js.org/2021/02/06/wechaty-getting-started-without-leave-your-browser/)

Reviewer: ```suggestion
Gitpod is an online and open-source platform for automated and ready-to-code development environments. If you have never used Gitpod before, you will be required to log in using your GitHub account.
You can learn more about Gitpod  Wechaty from our blog: [Getting Started Without Leaving Your Browser: Wechaty Gitpod, @huan, Feb 06, 2021](https://wechaty.js.org/2021/02/06/wechaty-getting-started-without-leave-your-browser/)
```

#### Using Google Cloud Shell

[![Open in Cloud Shell][shell_img]][shell_link]
[shell_img]: https://gstatic.com/cloudssh/images/open-btn.svg
[shell_link]: https://ssh.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https%3A%2F%2Fgithub.com%2Fwechaty%2Fwechaty-getting-started&cloudshell_open_in_editor=examples/ding-dong-bot.ts&cloudshell_workspace=.&cloudshell_tutorial=examples/tutorials/google-cloud-shell-tutorial.md

Google Cloud Shell is an online development and operations environment accessible anywhere with your browser. You can run this project on Google Cloud Shell by clicking the button below.
Generated via [open-in-cloud-shell](https://cloud.google.com/shell/docs/open-in-cloud-shell)

> After opening the Google Cloud Shell editor, there  it should open  a tutorial in the right panel which you can follow to learn more about Wechaty.

Reviewer: ```suggestion
After opening the Google Cloud Shell editor, there should open a tutorial in the right panel which you can follow to learn more about Wechaty.
```

<PolyglotTable />

:::tip
## Wechaty Getting Started Project

We have the world's shortest lines code of chatbot in all programming languages!
If you are a total beginner to Wechaty, this project is the best starting point for you. You can run it on a Cloud IDE in a couple of steps or locally on your machine.
**Features of Wechaty Getting Started project**

:::
* It works in all operating systems like Linux, Mac or Windows.
* It Supports all puppets like Web, Pad, Windows, and Mac.
* It replies with a `dong` message whenever it receives a `ding` message.

> <ShortestChatbots />

Reviewer 2: I'd like to keep this code example in this overview docs.

Or could you please let me know why you believe it should be removed?

Reviewer 2: Could we keep this?

Author: @huan  done

### Running Wechaty Getting Started Project on a cloud-based IDE

:::tip
The fastest way to get started with Wechaty is to use a Cloud-based IDE for running the Wechaty Getting Started Project. You can either use Gitpod or Google Cloud Shell.
If you are a total beginner, then we recommend Gitpod.

Wechaty in all programming languages have the same API, as the TypeScript documented version.
#### Using Gitpod Wechaty

:::
[![GitPod Ready-to-Code][gitpod_img]][gitpod_link]

> ## Getting Started Templates {#getting-started-templates}

Reviewer 2: The `polyglot` docs are under the **references** section, which means that they are technical descriptions of the machinery and how to operate it.

they should be information-oriented, austere, and to the point. <https://documentation.divio.com/reference/>

So I believe the `getting started` part in this docs is not necessary.

Please feel free to let me know if you have any questions.

Author: yes @huan as the current polyglot version started a bit abruptly hence just gave a bit of intro and gave links to tutorials page for info

- [Multi Language Wechaty Beta Release Announcement!](https://wechaty.js.org/2020/06/19/multi-language-wechaty-beta-release/)
Google Cloud Shell is an online development and operations environment accessible anywhere with your browser. You can run this project on Google Cloud Shell by clicking the button below.Generated via [open-in-cloud-shell](https://cloud.google.com/shell/docs/open-in-cloud-shell)
After opening the Google Cloud Shell editor, there should open a tutorial in the right panel which you can follow to learn more about Wechaty.
Learn more about running this project on Google Cloud Shell from [Google Cloud Shell Tutorials for Wechaty](https://wechaty.js.org/docs/quick-start/running-on-google-cloud-shell)

## Blogs

## Docs (Issues)
Here are some of the blogs related to latest Wechaty Beta releases.

- [Wechaty is All You Need: Python, Go, and Java Translation Project wechaty/wechaty#1927](https://github.com/wechaty/wechaty/discussions/1927)

> - [Python the Wechaty Way wechaty/python-wechaty#13](https://github.com/wechaty/python-wechaty/issues/13)

Reviewer 2: Please keep those links because they are very informative and helpful.

Reviewer 2: Could we keep this?

Reviewer 2: Could we keep this???

<PolyglotTable />

:::tip
## Wechaty Polyglot

We have the world's shortest lines code of chatbot in all programming languages!
Wechaty has the world's shortest lines code of chatbot in all programming languages!.This section gives an overview of implementing in different programming languages

:::
If you are a total beginner to Wechaty, this project is the best starting point for you. You can run it on a Cloud IDE in a couple of steps or locally on your machine.
**Features of Wechaty Getting Started project**

<ShortestChatbots />
* It works in all operating systems like Linux, Mac or Windows.
* It Supports all puppets like Web, Pad, Windows, and Mac.
* It replies with a `dong` message whenever it receives a `ding` message.

:::tip
### Running Wechaty Getting Started Project on a cloud-based IDE

> Wechaty in all programming languages have the same API, as the TypeScript documented version.

Reviewer 2: This sentence is import, please keep it.

Author: @huan  yes it is mentioned in all the different languages docs , only here it is removed to avoid repetition

The fastest way to get started with Wechaty is to use a Cloud-based IDE for running the Wechaty Getting Started Project. You can either use Gitpod or Google Cloud Shell.
If you are a total beginner, then we recommend Gitpod.

:::
#### Using Gitpod Wechaty

## Getting Started Templates {#getting-started-templates}
[![GitPod Ready-to-Code][gitpod_img]][gitpod_link]

> <GettingStartedTemplates />

Reviewer 2: Why do we remove the `<GettingStartedTemplates />`? 

Could you please share your thoughts about this change, because I think it's an important part for the readers?

Reviewer 2: Could we keep this?

Author: @huan  yes have added it

> #### Using Google Cloud Shell

Reviewer 2: The `polyglot/` docs are under **References**, which means that it should be straightforward, information-oriented, austere, and to the point.

> Reference guides are technical descriptions of the machinery and how to operate it. --- [Documentation System](https://documentation.divio.com/reference/)

I think we can remove all the getting-started related information in this docs, hence the "Using Google Cloud Shell" is not necessary.

Author: ok will do so

Author: @huan  have removed all the "Using Google Cloud Shell" part from the doc as suggested

[gitpod_img]: https://img.shields.io/badge/Gitpod-Ready--to--Code-blue?logo=gitpod
[gitpod_link]: https://gitpod.io/#https://github.com/wechaty/wechaty-getting-started

## Blogs
Gitpod is an online and open-source platform for automated and ready-to-code development environments. If you have never used Gitpod before, you will be required to log in using your GitHub account.
You can learn more about Gitpod  Wechaty from [Gitpod Tutorials for Wechaty](https://wechaty.js.org/docs/quick-start/running-on-gitpod).

#### Using Google Cloud Shell

[![Open in Cloud Shell][shell_img]][shell_link]
[shell_img]: https://gstatic.com/cloudssh/images/open-btn.svg
[shell_link]: https://ssh.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https%3A%2F%2Fgithub.com%2Fwechaty%2Fwechaty-getting-started&cloudshell_open_in_editor=examples/ding-dong-bot.ts&cloudshell_workspace=.&cloudshell_tutorial=examples/tutorials/google-cloud-shell-tutorial.md

> - [Multi Language Wechaty Beta Release Announcement!](https://wechaty.js.org/2020/06/19/multi-language-wechaty-beta-release/)

Reviewer 2: The release announcement is import for this docs, please keep it.

Author: yes it is kept
